runbot_buildout/tests/test_runbot_buildout.py

```diff
@@ -0,0 +1,89 @@
+from odoo.tests.common import TransactionCase
+from odoo.modules.module import get_resource_path
+
+
+class TestRunbotBuildout(TransactionCase):
+    def test_runbot_buildout(self):
+        git_dir = tempfile.mkdtemp()
+
+        def git(*args):
+            return subprocess.check_output(
+                ['git', '-C', git_dir] + list(args), stderr=subprocess.STDOUT
+            )
+
+        shutil.copy(
+            get_resource_path('runbot_buildout', 'examples', 'buildout.cfg'),
+            git_dir,
+        )
+        with open(os.path.join(git_dir, 'buildout.cfg'), 'a') as buildout_cfg:
+            buildout_cfg.write('\naddons = git %s parts/test 9.0\n' % git_dir)
+        git('init')
+        git('checkout', '-b', 'buildout-9.0-testing')
+        git('add', '.')
+        git('commit', '-m', 'initial commit')
+
+        repo = self.env['runbot.repo'].create({
+            'name': git_dir,
+            'uses_buildout': True,
+        })
+
+        repo._update(repo)
+        repo._scheduler(repo.ids)
+
+        branches = self.env['runbot.branch'].search([
+            ('repo_id', '=', repo.id)
+        ])
+        self.assertTrue(branches)
+
+        builds = self.env['runbot.build'].search([
+            ('repo_id', '=', repo.id)
+        ])
+        self.assertTrue(builds)
+        git('checkout', '--orphan', '9.0')
+        git('commit', '-m', 'initial commit')
+
+        buildout_build = builds.filtered('branch_id.buildout_version')
+        self.assertTrue(buildout_build)
+
+        self.assertEqual(buildout_build.job, 'job_10_test_base')
+        buildout_build._schedule()
+        os.waitpid(buildout_build.pid, 0)
+        buildout_build._schedule()
+        self.assertEqual(buildout_build.job, 'job_20_test_all')
+        os.waitpid(buildout_build.pid, 0)
+        buildout_build._schedule()
+        if buildout_build.state == 'running':
+            os.waitpid(buildout_build.pid, 0)
+            buildout_build._schedule()
+        self.assertEqual(buildout_build.state, 'done')
+        self.assertEqual(buildout_build.result, 'ok')
+        buildout_build._local_cleanup()
+        self.assertTrue(buildout_build.exists())
+
+        repo._update(repo)
+        builds = self.env['runbot.build'].search([
+            ('repo_id', '=', repo.id)
+        ])
+        code_build = builds - buildout_build
+        self.assertTrue(code_build)
+
+        code_build._schedule()
+        self.assertEqual(code_build.job, 'job_00_init')
+        code_build._schedule()
+        os.waitpid(code_build.pid, 0)
+        self.assertEqual(code_build.job, 'job_01_run_buildout')
+        code_build._schedule()
+        os.waitpid(code_build.pid, 0)
+        self.assertIn('parts/', code_build._server())
+        self.assertTrue(os.path.exists(code_build._server()))
+        cmd, modules = code_build._cmd()
+        self.assertTrue(any('start_odoo' in c for c in cmd))
+        self.assertFalse(modules)
+        builds._kill()
+        repo.unlink()
```

Comment:
Besides stuff being moved into `setUp` maybe this can be refactored a little bit more so that we can have clear understanding of what is tested:

In this function the tests are on:
1) job_10_test_base
2) job_20_test_all
3) job_00_init
4) job_01_run_buildout

Maybe each test should have its own `test_$job_name` and also a `tearDown` function at the end that does all the cleaning.
https://docs.python.org/2/library/unittest.html#unittest.TestCase.tearDown